flow2ml/Data_Loader.py

```diff
@@ -64,22 +86,23 @@ def create_dataset(self,classPath):
     for image in list(os.listdir(classPath)):
       
         if (image.find("Images") != -1):
-          # path is a folder
+          # path is a folder containing multiple preprocessed images
 
             for image_in_folder in (list(os.listdir(os.path.join(classPath,image)))):
               
 
               filtered_image = os.path.join(classPath,image,image_in_folder)
-              shutil.copy(filtered_image,final_data_folder)
-              
-              # Handle path seperators for Linux and Windows
-              if '/' in classPath:
-                folderName = list(classPath.split("/"))[2]
-              else:
-                folderName = os.path.split(classPath)[1]              
+              self.process_images(filtered_image, final_data_folder, image_in_folder, classPath)
+    
+    if not any([i != -1 for i in [image.find("Images") for image in os.listdir(classPath)]]):
 
-              ind = self.classes.index(folderName)
-              self.img_label[image_in_folder] = np.squeeze(np.eye(len(self.classes))[ind])
+      for image in list(os.listdir(classPath)):
+
+        if not(image.find("Images") != -1):
+        # path is an image since folder could not be found
+        
+          original_image = os.path.join(classPath,image)
+          self.process_images(original_image, final_data_folder, image, classPath)
```

Comment:
How do you get image label without returning anything from process_images()

Reply:
self.img_label is a class variable. So it just gets updated for each image everytime the function is called. 

This is what it looks like on printing. All image names and labels are correct:
![image](https://user-images.githubusercontent.com/50259869/119137214-30010f00-ba5e-11eb-8af7-6fcb0872c452.png)

Reply:
Right